flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/FunctionITCase.java

```diff
@@ -1698,6 +1717,32 @@ public void eval(@DataTypeHint("STRING") StringData s) {
         }
     }
 
+    /** This is an empty synchronous table function. */
+    private static class LookupTableLevel0Function<T> extends TableFunction<T> {}
+
+    /**
+     * Synchronous table function that uses {@link RowData} type inference for {@link
+     * LookupTableSource} and inherits {@link TableFunction} at multiple levels.
+     */
+    public static class LookupTableWithoutHintLevel1Function
```

Comment:
Just curious, should we add a test like LookupTableWithHintLevel1Function for considerations about the test orthogonality?

Reply:
I have added the `testLookupTableFunctionWithHintLevel1` test.